docs/src/pages/demos/dialogs/MaxWidthDialog.js

```diff
@@ -84,7 +86,7 @@ class MaxWidthDialog extends React.Component {
                     id: 'max-width',
                   }}
                 >
-                  <MenuItem value={false}>false</MenuItem>
+                  <MenuItem value="false">false</MenuItem>
```

Comment:
Why changing the type?

Reply:
This had to be changed because `MenuItem` only accepts `string | number | string[] | undefined` for its type. The change in the `handleMaxWidthChange` covers converting this back to a `boolean` when this value is chosen!

Reply:
Could MenuItem accept `boolean` too?

Reply:
Should be addressed in #15245.

Reply:
:weary: We're breaking quite a bit of DOM/type safety for some tiny conveniences. We could just use an empty string and check for that. Otherwise we need to override the actual DOM types.

Reply:
Is that loss of type safety from overriding the defaul DOM types?

Reply:
It's passed straight down to the DOM node, yes. It's probably better to override this with `unknown` since it is also used as a child of `Select`. But it's also used without the `Select` context. In different contexts you might be missing the type checking. I think we should use either some type casting or add the runtime overhead that ensures a sound `value` type. The value of `MenuItem` should not be changed.

Reply:
So should we keep the value of MenuItem as false and cast it to unknown?

Also, will there be issues trying to read event.target.value?

Reply:
> event.target.value

They always end up as strings in the the DOM. With the exception of `Radio` and `RadioGroup` no component reads this value though. We try to only use the value from props.

> So should we keep the value of MenuItem as false and cast it to unknown?

I think the `Select` already passes the `value` as the second argument to `onChange`. That `value` is the value from the prop and not the DOM. You could use that instead and just force it into state via `any`/`unknown`.

Reply:
ok I see. So the idea is to mostly use the `value` returned as the second param in the `onChange` event? I will push a change!

Reply:
Well it depends. If you want the value you passed as a prop you use the second param (type `unknown`). If you want the value in the DOM you use `event.target.value` (type `string`).